drivers/can/can_mcux_flexcan.c

```diff
@@ -811,31 +811,32 @@ static const struct can_driver_api mcux_flexcan_driver_api = {
 DT_INST_FOREACH_STATUS_OKAY(FLEXCAN_DEVICE_INIT_MCUX)
 
 #if defined(CONFIG_NET_SOCKETS_CAN)
-#include "socket_can_generic.h"						\
-#define FLEXCAN_DEVICE_SOCKET_CAN(id)					\
-	static int socket_can_init_##id(const struct device *dev)	\
-	{								\
-		struct device *can_dev = DEVICE_DT_INST_GET(id);	\
-		struct socket_can_context *socket_context = dev->data;	\
-		LOG_DBG("Init socket CAN device %p (%s) for dev %p (%s)", \
-			dev, dev->name, can_dev, can_dev->name);	\
-		socket_context->can_dev = can_dev;			\
-		socket_context->msgq = &socket_can_msgq;		\
-		socket_context->rx_tid =				\
-		k_thread_create(&socket_context->rx_thread_data,	\
-				rx_thread_stack,			\
-				K_KERNEL_STACK_SIZEOF(rx_thread_stack),	\
-				rx_thread, socket_context, NULL, NULL,	\
-				RX_THREAD_PRIORITY, 0, K_NO_WAIT);	\
-		return 0;						\
-	}								\
-									\
-	NET_DEVICE_INIT(socket_can_flexcan_##id, SOCKET_CAN_NAME_##id,	\
-		socket_can_init_##id, device_pm_control_nop,		\
-		&socket_can_context_##id, NULL,				\
-		CONFIG_KERNEL_INIT_PRIORITY_DEVICE, &socket_can_api,	\
-		CANBUS_RAW_L2, NET_L2_GET_CTX_TYPE(CANBUS_RAW_L2),	\
-		CAN_MTU);						\
-
-DT_INST_FOREACH_STATUS_OKAY(FLEXCAN_DEVICE_SOCKET_CAN)
+
+#include "socket_can_generic.h"
+
+static int socket_can_init_1(const struct device *dev)
+{
+	const struct device *can_dev = DEVICE_DT_GET(DT_NODELABEL(flexcan0));
```

Comment:
Not that I use SocketCAN, but it's unfortunate that we bind this hard to `flexcan0`.

Reply:
Building on that a bit, does the use of `flexcan0` take a step backward from recent update https://github.com/zephyrproject-rtos/zephyr/commit/0519a6a0e794343b6c078d4971866fa79fa74742?

Reply:
changed to use zephyr,can-primary label, could specify the detail can device by dts overlay file